lib/logstash/outputs/scalyr.rb

```diff
@@ -215,22 +239,24 @@ def multi_receive(events)
     total_batches = multi_event_request_array.length unless multi_event_request_array.nil?
 
     result = []
+    had_records = !multi_event_request_array.to_a.empty?
```

Comment:
We could also move this check to the begining of this function and just return therre directly in case the batch is empty.

Reply:
That is what I had earlier and if we do that we won't send status messages when there is no events, which might result in misleading people into thinking the plugin is broken (as it did me for a bit there). Basically we need to reach `send_status` every loop. Could duplicate the call to it at the top of the method but I don't know whats better style wise.

Reply:
ack, thanks for the clarification. should be fine then.